test/persist/test_failure_indices.sh

```diff
@@ -0,0 +1,124 @@
+#! /bin/bash
```

Comment:
we should set stricter modes on this: errexit, pipefail, etc.

Reply:
Specifically: 

```suggestion
#!/bin/bash
set -euo pipefail
```